### PR TITLE
Update NucleoATAC to version 0.3.4

### PR DESCRIPTION
**Pull request by lparsons**
Added imports test, which forces a build
Add fix for zlib.h not found
Preserve nucleoatac version 0.3.1

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/nucleoatac/meta.yaml [hunk 1]
```diff
@@ -1,20 +1,20 @@
 {% set name = "nucleoatac" %}
-{% set version = "0.3.1" %}
+{% set version = "0.3.4" %}
 
 package:
-  name: {{ name|lower }}
-  version: {{ version }}
-  
+  name: nucleoatac
```
Comment (bgruening) [inline recipes/nucleoatac/meta.yaml]: @lparsons can we keep the template-vars here?

**Comment by lparsons** [comment]
@bgruening Sure thing, preserved template-vars.

File: recipes/nucleoatac/meta.yaml [hunk 2]
```diff
@@ -28,15 +28,12 @@ requirements:
...
     - scipy
     - matplotlib
     - pysam >=0.8.1
-    
-test:
-  commands:
-    - nucleoatac run --help
-    - pyatac --help
\ No newline at end of file
+    - zlib
```
Comment (bgruening) [inline recipes/nucleoatac/meta.yaml]: This looks also wrong. Why zlib here? It should be a run dependency, but giving that the test pass?

Comment (lparsons) [reply recipes/nucleoatac/meta.yaml]: Sorry, I'm not sure I follow. Yout think it should, or should not, be a run dependency? My issue with this package was that is builds some cython code when you attempt to import, for example, `pyatac.fragmentsizes`. This compilation was failing under the old version and wasn't tested. My intention here was to ensure the test actually does something other than display help (which doesn't properly test dependencies). Really, this should probably be done during `setup.py`, but that's an upstream thing I didn't want to get into just now.

Comment (bgruening) [reply recipes/nucleoatac/meta.yaml]: Oh, sorry, I just saw this wrong on my mobile. You have actually removed the test commands here. Can you please get them back :)
Also please pin zlib as well.
Thanks @lparsons!

**Comment by lparsons** [comment]
The attempt was to move them to the test script, with   some additional tests. I've had issues with this in the past (the docs are confusing on the subject), is the test script not running?

--
Lance Parsons - Scientific Programmer
Carl C. Icahn Laboratory - Room 136
Lewis-Sigler Institute for Integrative Genomics
Princeton University




On Fri, Sep 29, 2017 at 5:17 PM -0400, "Björn Grüning" <notifications@github.com<mailto:notifications@github.com>> wrote:


@bgruening commented on this pull request.

________________________________

In recipes/nucleoatac/meta.yaml<https://github.com/bioconda/bioconda-recipes/pull/6177#discussion_r141970555>:

> \ No newline at end of file
+    - zlib


Oh, sorry, I just saw this wrong on my mobile. You have actually removed the test commands here. Can you please get them back :)
Also please pin zlib as well.
Thanks @lparsons<https://github.com/lparsons>!

File: recipes/nucleoatac/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# For zlib: https://bioconda.github.io/troubleshooting.html#zlib
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
+export CPATH=${PREFIX}/include
+
```
Comment (bgruening) [inline recipes/nucleoatac/run_test.sh]: Can I propose a different solution?
Can you remove this test script and use plain `commands` and python module tests:

You export CPATH, you please put in the build.sh script and zlib should be in the run and build environment.

**Comment by lparsons** [comment]
I'll try again, but the import actually does the build, so I ran into problems with with zlib.h during the import without the env vars set.

--
Lance Parsons - Scientific Programmer
Carl C. Icahn Laboratory - Room 136
Lewis-Sigler Institute for Integrative Genomics
Princeton University




On Sat, Sep 30, 2017 at 8:29 AM -0400, "Björn Grüning" <notifications@github.com<mailto:notifications@github.com>> wrote:


@bgruening commented on this pull request.

________________________________

In recipes/nucleoatac/run_test.sh<https://github.com/bioconda/bioconda-recipes/pull/6177#discussion_r142005179>:

> @@ -0,0 +1,11 @@
+#!/bin/bash
+
+# For zlib: https://bioconda.github.io/troubleshooting.html#zlib
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
+export CPATH=${PREFIX}/include
+


Can I propose a different solution?
Can you remove this test script and use plain commands and python module tests:

You export CPATH, you please put in the build.sh script and zlib should be in the run and build environment.

**Comment by bgruening** [comment]
Very nice! Thanks Lance!

**Pull request merged**